devices/SPS30/SHDLC/SHDLCProtocol.cs

```diff
@@ -0,0 +1,177 @@
+﻿//
+// Copyright (c) 2017 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Ports;
+
+namespace Iot.Device.SPS30.SHDLC
+{
+    /// <summary>
+    /// Minimal implementation of the SHDLC protocol used by Sensirion. This implementation is based upon the information
+    /// available in the SPS30 particulate matter sensor datasheet.
+    /// </summary>
+    /// <remarks>
+    /// Datasheet can be found at https://sensirion.com/media/documents/8600FF88/616542B5/Sensirion_PM_Sensors_Datasheet_SPS30.pdf
+    /// Information about SHDLC can be found at https://sensirion.github.io/python-shdlc-driver/shdlc.html
+    /// </remarks>
+    public class SHDLCProtocol
+    {
+        /// <summary>
+        /// The SerialPort is exposed as public property, but should only be used in very specific circumstances. One example is the Sensirion SPS30, for which a raw "0xFF" must be sent to wake the device from its sleep.
```

Comment:
This particular device wants a 0xFF to wake the device, but it is unspecified in the SHDLC protocol (or rather, not mentioned it is more generic than only this device). Another device using the SHDLC protocol may divert from this. This is why the wake-up routine is part of the Sps30Sensor class instead of the SHDLC protocol implementation.

We expose the Serial for the sake of flexibility. We don't really need to shield the developer from this, since they have access to the underlying `SerialPort` object anyway.